innkeeper/client.go

```diff
@@ -320,6 +318,43 @@ func getHttpError(r *http.Response) (error, bool) {
 	}
 }
 
+func (c *Client) writeRoute(url string, route *routeData) error {
+
+	res, err := json.Marshal(route)
+
+	req, err := http.NewRequest("POST", url, bytes.NewReader(res))
+
+	if err != nil {
+		return err
+	}
+
+	authToken, err := c.opts.Authentication.GetToken()
```

Comment:
i think this is not right. This client accepts currently anything that implements the Authentication interface. It is true that it can be the FixedToken but it also can be anything else. For the integrity of the package, the package should make sure that it properly deals with what it offers.